B 14 change user password

shaguahill opened the pull request:
Created html file for change password
Added link to change-password page
Allowed url to SecurityFilter
Created ChangePasswordServlet
Added resolve bug fix for jetty
Created method changeUserPassword
Created method isOldPassword for check input old password
Added fields to edit user
Added method doPost to ChangePasswordServlet
Created change-password.css for change-password.html
Added tests

shaguahill committed:
[b-14] Change user password
Created html file for change password
Added link to change-password page
Allowed url to SecurityFilter
Created ChangePasswordServlet
Added resolve bug fix for jetty

shaguahill committed:
[b-14] Change user password
Created method changeUserPassword
Created method isOldPassword for check input old password
Added fields to edit user
Added method doPost to ChangePasswordServlet
Created change-password.css for change-password.html
Added tests

shaguahill committed:
Merge branch 'develop' into b-14_change_user_password

ValentynaBondarenko commented on src/main/java/com/photostudio/web/servlet/user/ChangePasswordServlet.java:
```diff
@@ -0,0 +1,64 @@
+                userService.changeUserPassword(user, newPassword);
+                response.sendRedirect(request.getContextPath() + "/user?id=" + user.getId());
+            } else {
+                paramsMap.put("invalid", "notMatchPassword");
```
можливо потрібно і сюди додати логи

shaguahill replied:
Added

Yuliia-Sh commented on src/main/webapp/WEB-INF/templates/user-info.html:
```diff
@@ -66,8 +66,9 @@ <h3 class="user__title-login">Login</h3>
+
+                    <a th:if="${isMe}" th:href="@{/security/change-password}" class="btn btn__manage btn__manage--pink btn__manage--mb" style="line-height: 40px">Change password
+                    </a>
 
```
Администратор разве не может изменить пароль пользователю?

shaguahill replied:
Вважаю, що ні, бо адміністратор не повинен знати пароль користувача(хоча він і має. в нашому випадку, доступ до коду та бази). Правильніше було б реалізувати функцію відновлення паролю самим користувачем, або на крайній випадок - щоб адмін мав доп кнопку, по якій буде генеруватись рандомний пароль і відправлятись на пошту користувачу. Але це питання я хочу обговорити з Мариною.

shaguahill committed:
[b-14] Change user password
Added pattern for new password: min 8 symbols and contain at list one number one lowercase letter and one uppercase letter

shaguahill committed:
Merge branch 'b-14_change_user_password' of https://github.com/Rooman/PhotoStudio into b-14_change_user_password

Yuliia-Sh commented on src/main/java/com/photostudio/dao/jdbc/JdbcUserDao.java:
```diff
@@ -185,8 +187,11 @@ public void edit(User user) {
+            preparedStatement.setLong(13, user.getId());
+            preparedStatement.setInt(14, user.getLangId());
+            preparedStatement.setLong(15, user.getId());
             preparedStatement.executeUpdate();
```
Для чего тут параметр №13?

shaguahill replied:
Дякую за уважність. Бага після виправлення конфліктів з девелопом. Виправлено

shaguahill committed:
[b-14] Change user password (review fix)

duplicate preparedStatement.setLong(13, user.getId()); deleted

shaguahill committed:
[b-14] Change user password (review fix)
Added logs

Yuliia-Sh commented on src/main/webapp/assets/css/change-password.css:
```diff
@@ -0,0 +1,89 @@
+footer {
+    bottom: 0;
+    width: 100%;
+}
```
footer - это ж фрагмент, если менять, то менять для всех страниц наверное нужно?
